multi-int-store/src/main/java/com/connexta/multiintstore/adaptors/S3StorageAdaptor.java

```diff
@@ -67,30 +75,29 @@ public void store(
         key);
   }
 
-  // TODO: Uncomment when the MIS implements retrieval
+  @Override
+  public RetrieveResponse retrieve(String key) throws StorageException {
+    log.info("Retrieving product in bucket \"{}\" with key \"{}\"", s3BucketQuarantine, key);
 
-  //  @Override
-  //  public RetrieveResponse retrieve(String ingestId) throws IOException {
-  //    log.info(
-  //        "Retrieving product in bucket \"{}\" with key \"{}\"", s3BucketQuarantine, ingestId);
-  //
-  //    final ResponseInputStream<GetObjectResponse> getObjectResponseResponseInputStream =
-  //        s3Client.getObject(
-  //            GetObjectRequest.builder().bucket(s3BucketQuarantine).key(ingestId).build());
-  //    final GetObjectResponse getObjectResponse = getObjectResponseResponseInputStream.response();
-  //
-  //    return new RetrieveResponse(
-  //        MediaType.valueOf(getObjectResponse.contentType()),
-  //        new ByteArrayResource(getObjectResponseResponseInputStream.readAllBytes()) {
-  //          @Override
-  //          public String getFilename() {
-  //            final Map<String, String> metadata = getObjectResponse.metadata();
-  //            final String filename = metadata.get("filename");
-  //            if (StringUtils.isEmpty(filename)) {
-  //              return ingestId;
-  //            }
-  //            return filename;
-  //          }
-  //        });
-  //  }
+    try (final ResponseInputStream<GetObjectResponse> getObjectResponseResponseInputStream =
+        s3Client.getObject(
+            GetObjectRequest.builder().bucket(s3BucketQuarantine).key(key).build())) {
+      final GetObjectResponse getObjectResponse = getObjectResponseResponseInputStream.response();
+      return new RetrieveResponse(
+          MediaType.valueOf(getObjectResponse.contentType()),
+          new ByteArrayResource(getObjectResponseResponseInputStream.readAllBytes()) {
```

Comment:
:exclamation: readAllBytes read the entire stream into memory. This is fine for metadata because metadata should never be more than a few hundred K. However, the product is another matter. It could be multiple gigabytes.

Reply:
I'll ask around to see how we want to handle this. In any case, I don't think this is a blocker for this PR. We can address in a future PR because this PR is blocking other tasks.

Reply:
I talked with Eric and pushed a new commit to address this.